src/utils/miopen.cpp

```diff
@@ -0,0 +1,1506 @@
+  void* error_signal,
+  size_t ws_size,
+  void* ws) {
+  constexpr int num_trials = 3;
+  constexpr int num_skip = 1;
+  int num_algos = ConvolutionBwdDataAlgoCount();
+  std::vector<miopenConvAlgoPerf_t> perf_results_all;
+  std::vector<miopenConvAlgoPerf_t> perf_results(num_algos);
+  for (int trial = 0; trial < num_trials + num_skip; ++trial) {
+    int num_tested_algos;
+    CHECK_MIOPEN(miopenFindConvolutionBackwardDataAlgorithm(
+                  get_handle(),
+                  prev_error_signal_desc, prev_error_signal,
+                  kernel_desc, kernel,
+                  conv_desc,
+                  error_signal_desc, error_signal,
+                  num_algos, &num_tested_algos,
+                  perf_results.data(),
+                  ws, ws_size,
+                  false));
+    if (trial > num_skip) {
+      for (const auto& p : perf_results) {
+        perf_results_all.push_back(p);
+      }
+    }
+  }
+  return find_best_algorithm(perf_results_all, nondet_bwd_data_algos,
+                             deterministic, ws_size);
+  //return miopenConvolutionBwdDataAlgoGEMM;
+}
+
+miopenConvBwdWeightsAlgorithm_t get_bwd_filter_algo_autotune(
+  bool deterministic,
+  const TensorDescriptor& input_desc,
+  const void* input,
+  const TensorDescriptor& prev_error_signal_desc,
+  const void* prev_error_signal,
+  const ConvolutionDescriptor& conv_desc,
+  const FilterDescriptor& kernel_gradient_desc,
+  void* kernel_gradient,
+  size_t ws_size,
+  void* ws) {
+  constexpr int num_trials = 3;
+  constexpr int num_skip = 1;
+  int num_algos = ConvolutionBwdFilterAlgoCount();
+  std::vector<miopenConvAlgoPerf_t> perf_results_all;
+  std::vector<miopenConvAlgoPerf_t> perf_results(num_algos);
+  for (int trial = 0; trial < num_trials + num_skip; ++trial) {
+    int num_tested_algos;
+    CHECK_MIOPEN(miopenFindConvolutionBackwardWeightsAlgorithm(
+                  get_handle(),
+                  prev_error_signal_desc, prev_error_signal,
+                  input_desc, input,
+                  conv_desc,
+                  kernel_gradient_desc, kernel_gradient,
+                  num_algos, &num_tested_algos,
+                  perf_results.data(),
+                  ws, ws_size,
+                  false));
+    if (trial > num_skip) {
+      for (const auto& p : perf_results) {
+        perf_results_all.push_back(p);
+      }
+    }
+  }
+  return find_best_algorithm(perf_results_all, nondet_bwd_filter_algos,
+                             deterministic, ws_size);
+  //return miopenConvolutionBwdWeightsAlgoGEMM;
+}
+
+}  // namespace
+
+fwd_conv_alg get_fwd_algorithm(
+  bool autotune,
+  bool deterministic,
+  const TensorDescriptor& input_desc,
+  const void* input,
+  const FilterDescriptor& kernel_desc,
+  const void* kernel,
+  const ConvolutionDescriptor& conv_desc,
+  const TensorDescriptor& output_desc,
+  void* output,
+  size_t ws_size,
+  void* ws) {
+  return fwd_conv_alg::IMPLICIT_GEMM;
```

Comment:
Is this debugging code?
```suggestion
```

Reply:
good catch, these are indeed from debugging. I'll remove them and test the install again!